app/devices/index.js

```diff
@@ -50,4 +84,160 @@ module.exports.turnOffDevice = functions.https.onRequest((req, res) => {
     }
     turnOffDevice(res,req);
   });
+});
+
+
+
+module.exports.device = functions.region('europe-west1').https.onRequest((req, res) => {
+   return cors(req, res, () => {
+       if (req.method === 'GET') {
+           var devicesRef = dbref.child('Devices');
+           devicesRef.on('value', function (snapshot) {
+               return res.status(200).json(
+                   snapshot.child(req.body.id).val()
+               );
+           });
+       } else if (req.method === 'POST') {
+           return dbref.child('Devices/' + req.body.id).update({'enabled': req.body.enabled})
+               .then(res.status(200).json({
+                   'enabled': req.body.enabled
+               }))
+               .catch((err) => {
+                       console.log('Error updating database', err);
+                   }
+               );
+       }
+   })
+
+});
+
+module.exports.device_url = functions.region('europe-west1').https.onRequest((req, res) => {
+    return cors(req, res, () => {
+        if (req.method === 'GET') {
+            var devicesRef = dbref.child('Devices');
+            devicesRef.on('value', function (snapshot) {
+                return res.status(200).json(snapshot.child(req.query.id).val());
+            });
+        } else if (req.method === 'POST') {
+            return dbref.child('Devices/' + req.query.id).update({ 'enabled': req.query.enabled }).then(res.status(200).json({
+                'enabled': req.query.enabled
+            })).catch(err => {
+                console.log('Error updating database', err);
+            });
+        }
+    });
+});
+
+const updateDeviceStatus = (res, req) => {
+    return dbref.child('Devices/' + req.query.id).update({'enabled': req.query.enabled})
+        .then(res.status(200).json({
+            'enabled': req.query.enabled
+        }))
+        .catch((err) => {
+                console.log('Error updating database', err);
+            }
+        );
+};
+
+/**
+ * Send a POST request such as
+ * https://europe-west1-smarthome-3c6b9.cloudfunctions.net/updateDeviceStatus?id=my9iXu6WvEgx5oNLLegs&enabled=true
+ * to confirm that the device is turned on, and
+ * https://europe-west1-smarthome-3c6b9.cloudfunctions.net/updateDeviceStatus?id=my9iXu6WvEgx5oNLLegs&enabled=false
+ * to confirm that the device is turned off
+ * @type {HttpsFunction}
+ */
+module.exports.updateDeviceStatus = functions.region('europe-west1').https.onRequest((req, res) => {
```

Comment:
Alright, thanks. I made it that way because it was easier to use for the web unit guys, but I see that  everything should ideally be done through JSON